devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy

```diff
@@ -1,300 +1,531 @@
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+    log.debug "BatteryReport"
+    log.debug "cmd: "+cmd
+    log.debug "location: "+location
+    def timeDate = location.timeZone ? new Date().format("yyyy MMM dd EEE h:mm:ss a", location.timeZone) : new Date().format("yyyy MMM dd EEE h:mm:ss")
+    //def timeDate = new Date().format("yyyy MMM dd EEE HH:mm:ss", location.timeZone)
+    sendEvent(name: "battery", value: cmd.batteryLevel)
+    sendEvent(name: "batteryStatus", value: "Battery: $cmd.batteryLevel%\n($timeDate)")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+    log.debug "NotificationReport"
+    def map = [:]
+    def alarmInfo = "Last alarm detection: "
+    if (cmd.notificationType == 5) {
+        switch (cmd.event) {
+            case 2:
+                map.name = "water"
+                map.value = "wet"
+                map.descriptionText = "${device.displayName} is ${map.value}"
+                state.lastAlarmDate = "\n"+new Date().format("yyyy MMM dd EEE HH:mm:ss")
+                //state.lastAlarmDate = "\n"+new Date().format("yyyy MMM dd EEE HH:mm:ss", location.timeZone)
+                multiStatusEvent(alarmInfo + state.lastAlarmDate)
+                break
+
+            case 0:
+                map.name = "water"
+                map.value = "dry"
+                map.descriptionText = "${device.displayName} is ${map.value}"
+                multiStatusEvent(alarmInfo + state.lastAlarmDate)
+                break
+        }
+    } else if (cmd.notificationType == 7) {
+        switch (cmd.event) {
+            case 0:
+                map.descriptionText = "${device.displayName}: tamper alarm has been deactivated"
+                sendEvent(name: "tamper", value: "clear")
+                sendEvent(name: "batteryStatus", value: "Tamper alarm inactive")
+                break
+
+            case 3:
+                map.descriptionText = "${device.displayName}: tamper alarm activated"
+                sendEvent(name: "tamper", value: "detected")
+                sendEvent(name: "batteryStatus", value: "Tamper alarm activated")
+                break
+        }
+    }
+    createEvent(map)
+}
+
+private multiStatusEvent(String statusValue, boolean force = false, boolean display = false) {
+    if (!device.currentValue("multiStatus")?.contains("Sync") || device.currentValue("multiStatus") == "Sync OK." || force) {
+        sendEvent(name: "multiStatus", value: statusValue, descriptionText: statusValue, displayed: display)
+    }
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+    log.debug "SensorMultilevelReport"
+    def tempValue
+    def map = [:]
+    if (cmd.sensorType == 1) {
+        // temperature
+        def cmdScale = cmd.scale == 1 ? "F" : "C"
+        tempValue = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmdScale, 1)

```

Comment:
confused about why two identical variables are needed here

Reply:
Change has been made. Before I update the PR I will have to run a test to make sure.